samples/csharp/getting-started/Ranking_AutoML/Ranking/Ranking/Program.cs

```diff
@@ -0,0 +1,234 @@
+            {
+                Separators = new[] { '\t' },
+                HasHeader = true,
+                Columns = new[]
+                {
+                    new TextLoader.Column("Label", DataKind.Single, 0),
+                    new TextLoader.Column("GroupId", DataKind.Int32, 1),
+                    new TextLoader.Column("Features", DataKind.Single, 2, 133),
+                }
+            };
+
+            TextLoader textLoader = mlContext.Data.CreateTextLoader(textLoaderOptions);
+            IDataView trainDataView = textLoader.Load(TrainDatasetPath);
+            IDataView validationDataView = textLoader.Load(ValidationDatasetPath);
+            IDataView testDataView = textLoader.Load(TestDatasetPath);
+
+            // STEP 2: Display first few rows of training data
+            ConsoleHelper.ShowDataViewInConsole(mlContext, trainDataView);
+
+            // STEP 3: Initialize our user-defined progress handler that AutoML will 
+            // invoke after each model it produces and evaluates.
+            var progressHandler = new RankingExperimentProgressHandler();
+
+            // STEP 4: Run AutoML ranking experiment
+            ConsoleHelper.ConsoleWriteHeader("=============== Running AutoML experiment ===============");
+            Console.WriteLine($"Running AutoML ranking experiment for {ExperimentTime} seconds...");
+
+            var experimentSettings = new RankingExperimentSettings
+            {
+                MaxExperimentTimeInSeconds = ExperimentTime,
+                OptimizingMetric = RankingMetric.Ndcg,
+                OptimizationMetricTruncationLevel = 10
+            };
+
+            ExperimentResult<RankingMetrics> experimentResult = mlContext.Auto()
+                .CreateRankingExperiment(experimentSettings)
+                .Execute(
+                    trainData: trainDataView,
+                    validationData: validationDataView,
+                    progressHandler: progressHandler);
+
+            // Print top models found by AutoML
+            Console.WriteLine();
+            PrintTopModels(experimentResult);
+
+            // Re-fit best pipeline on train and validation data, to produce 
+            // a model that is trained on as much data as is available while
+            // still having test data for the final estimate of how well the
+            // model will do in production.
+            Console.WriteLine("\n===== Refitting on train+valid and evaluating model's nDCG with test data =====");
+            var trainPlusValidationDataView = textLoader.Load(new MultiFileSource(TrainDatasetPath, ValidationDatasetPath));
+
+            var refitModel = experimentResult.BestRun.Estimator.Fit(trainPlusValidationDataView);
+
+            IDataView predictionsRefitOnTrainPlusValidation = refitModel.Transform(validationDataView);
+
+            // Setting the DCG truncation level
+            var rankingEvaluatorOptions = new RankingEvaluatorOptions { DcgTruncationLevel = 10 };
```

Comment:
Incase they choose a larger `OptimizationMetricTruncationLevel` above, ensure we produce the requested `@N`:
```suggestion
            var rankingEvaluatorOptions = new RankingEvaluatorOptions
                {
                    DcgTruncationLevel = Math.Min(10, experimentSettings.OptimizationMetricTruncationLevel) 
                };
```

Reply:
#Resolved